### DISCUSSION TITLE
useField's getInputProps

### DISCUSSION THREAD

**Discussion by marksmall** [airjp73/rvf#29] [Q&A] [answered]
I have followed the guide in the README to create an re-usable `Input` and `Button` components. I have in my `package.json`:

```
"@remix-validated-form/with-zod": "^1.0.1",
"remix-validated-form": "^3.1.1"
```

When I try to destructure the `useField` e.g.
```
const { error, getInputProps } = useField(name);
```
The editor states `Property 'getInputProps' does not exist on type 'FieldProps'.ts(2339)` and indeed, the form is broken. I found another blog somewhere, can't remember now where and instead, it uses:
```
const { validate, clearError, defaultValue, error } = useField(name);
```
I now use this as it works, but I'd like to conform to what is expected really and the library docs say to do something that seems broken, for me. Which is really supposed to be used, am I making a newbie mistake? 

For now, the latter works and I'm sticking with it, but that isn't what the README docs states to use, so it's a bit confusing.

**Comment by airjp73** [comment, marked as answer]
Hi @marksmall!

`getInputProps` was added in [3.2.0](https://github.com/airjp73/remix-validated-form/releases/tag/remix-validated-form-v3.2.0). Could you try updating `remix-validated-form` and see if that fixes it?

**Reply by marksmall** [reply]
Sorry, should have thought of that first, will do.
